libraries/botbuilder-dialogs-adaptive-testing/src/httpRequestMocks/httpRequestSequenceMock.ts

```diff
@@ -0,0 +1,77 @@
+/**
+ * @module botbuilder-dialogs-adaptive-testing
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { HttpMethod } from 'botbuilder-dialogs-adaptive';
+import * as nock from 'nock';
+import * as parse from 'url-parse';
+import { HttpRequestMock } from './httpRequestMock';
+import { HttpResponseMock } from './httpResponseMock';
+import { SequenceResponseManager } from './sequenceResponseManager';
+
+export enum BodyMatchType {
+    Exact = 'Exact',
+    Partial = 'Partial',
+}
+
+export interface HttpRequestSequenceMockConfiguration {
+    method?: string | HttpMethod;
+    url?: string;
+    matchType?: string | BodyMatchType;
+    body?: string;
+    responses?: HttpResponseMock[];
+}
+
+export class HttpRequestSequenceMock extends HttpRequestMock implements HttpRequestSequenceMockConfiguration {
+    public static $kind = 'Microsoft.Test.HttpRequestSequenceMock';
+
+    public method: HttpMethod;
+    public url: string;
+    public matchType = BodyMatchType.Partial;
+    public body = '';
+    public responses: HttpResponseMock[] = [];
+
+    public setup(): void {
+        if (this.body && (this.method === HttpMethod.DELETE || this.method === HttpMethod.GET)) {
+            throw new Error("GET and DELETE don't support matching body!");
+        }
+        const response = new SequenceResponseManager(this.responses);
+        const url = parse(this.url);
+        let path = this.url.substr(url.origin.length);
```

Comment:
Any reason not to just use `parse(this.url).pathname`?

Reply:
I found `url.query` is not stable, it can be a string and sometimes it can also be an object, so I decided not using this combination.

Reply:
If you invoke parse with two parameters, the second a Boolean with value "true", query should always be an object. That's the way the Node standard library API works too.